### PR TITLE
Move SOF code out of Zephyr SOC area

### PR DESCRIPTION
**Pull request by finikorg**
Moving SOF initialization out of Zephyr intel_adsp SOC area.

### CODE REVIEW

**Comment by finikorg** [comment]
Needs this PR to work https://github.com/thesofproject/zephyr/pull/44

File: zephyr/CMakeLists.txt [hunk 1]
```diff
@@ -253,6 +253,7 @@ zephyr_library_sources(
 	edf_schedule.c
 	ll_schedule.c
 	schedule.c
+	init.c
```
Comment (lgirdwood) [inline zephyr/CMakeLists.txt:256]: Probably best if we use src/platform/name/platform.c here ?

Comment (finikorg) [reply zephyr/CMakeLists.txt:256]: @lgirdwood So we can include `src/platform/intel/cavs/platform.c` and also init.c for initialization?

Comment (lgirdwood) [reply zephyr/CMakeLists.txt:256]: yes, we should sue the platform.c (for each platform here), that we we dont need to pull it into Zephyr.

Comment (finikorg) [reply zephyr/CMakeLists.txt:261]: Added commit 7f05b90a3ed269798b2b737c36a8bcc3beb9b065

**Pull request merged**